### PR TITLE
Add missing NULL termination to the String::repeat function.

### LINKED ISSUE #41159: String.repeat and format error
**Godot version:** 3.2
**OS/device including version:** Mac OSX GLE3

**Issue description:**
Found a very strange bug that looks like a pointer into memory going awry. String.repeat used to create a format string results in random unwanted chars.

```
tool
extends EditorScript

func _run():
	var repeats = "%s".repeat(3) % ['a','b','c']
	assert(repeats == 'abc', repeats)
```

Fails and produces:
```
 <built-in>:6 - Assertion failed: abcX
 <built-in>:6 - Assertion failed: abcX
 <built-in>:6 - Assertion failed: abc@
 <built-in>:6 - Assertion failed: abc)/=
)/=H
 <built-in>:6 - Assertion failed: abc5
 <built-in>:6 - Assertion failed: abc
 <built-in>:6 - Assertion failed: ab
```

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes #41159

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

File: core/ustring.cpp [hunk 1]
```diff
@@ -3071,6 +3071,7 @@ String String::repeat(int p_count) const {
 	const CharType *src = this->c_str();
 
 	new_string.resize(length() * p_count + 1);
+	new_string[length() * p_count] = 0;
```
Comment (akien-mga) [inline core/ustring.cpp:3074]: Should we make it `'\0'` to be more explicit?

Comment (bruvzg) [reply core/ustring.cpp:3074]: Not sure it's not clear enough this way, and all other cases of setting terminating zero are using ` = 0;` as well.

Comment (akien-mga) [reply core/ustring.cpp:3074]: Alright, consistency is the most important :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**